---
title: "210318_HeatmapsOfRelevantGenes_Pollak"
author: "Erin Osborne Nishimura"
date: "5/16/2021"
output: html_document
---

# Analysis: Heatmaps of key gene signature sets

The purpose of this analysis is to assess how and whether published gene signature sets are changing the Pollak et al dataset. To do this, we will calculate the fold-enrichment of 
Here is the Pollak et al dataset:

  * Pollak J, Rai KG, Funk CC, Arora S, Lee E, Zhu J, Price ND, Paddison PJ, Ramirez JM, Rostomily RC. [Ion channel expression patterns in glioblastoma stem cells with functional and therapeutic implications for malignancy](https://doi.org/10.1371/journal.pone.0172884) _PLoS One_ **2017**; Mar 6;12(3):e0172884.

For this analysis, we used the following publications:

  * Creighton CJ. [A gene transcription signature of the Akt/mTOR pathway in clinical breast tumors](https://doi.org/10.1038/sj.onc.1210245). _Oncogene_ **2007**;26:4648–55.

  * Dry JR, Pavey S, Pratilas CA, Harbron C, Runswick S, Hodgson D, et al. [Transcriptional Pathway Signatures Predict MEK Addiction and Response to Selumetinib (AZD6244)](https://doi.org/10.1158/0008-5472.can-09-1577). _Cancer Res_ **2010**;70:2264–73.

  * Loboda A, Nebozhyn M, Klinghoffer R, Frazier J, Chastain M, Arthur W, et al. [A gene expression signature of RAS pathway dependence predicts response to PI3K and RAS pathway inhibitors and expands the population of RAS pathway activated tumors.](https://doi.org/10.1186/1755-8794-3-26) _Bmc Med Genomics_ **2010**;3:26.

  * Pratilas CA, Taylor BS, Ye Q, Viale A, Sander C, Solit DB, et al. [V600EBRAF is associated with disabled feedback inhibition of RAF–MEK signaling and elevated transcriptional output of the pathway.](https://doi.org/10.1073/pnas.0900780106) _Proc National Acad Sci_ **2009**;106:4519–24. 

  * Wagle M-C, Kirouac D, Klijn C, Liu B, Mahajan S, Junttila M, et al. [A transcriptional MAPK Pathway Activity Score (MPAS) is a clinically relevant biomarker in multiple cancer types](https://doi.org/10.1038/s41698-018-0051-4). _Npj Precis Oncol_ **2018**;2:7.

```{r, echo = FALSE}

# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install(version = "3.12")
# 
# library(BiocManager)
# BiocManager::install("ComplexHeatmap")
# I need to download X11 - XQuartz on Mac to use ComplexHeatmap


#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install(version = "3.12")

#BiocManager::install("clusterProfiler")


```



## Step 1: Load packages
```{r load packges, warning = FALSE, message = FALSE}

library(RColorBrewer)
library(ggplot2)
library(dplyr)
library(pheatmap)
library(reshape2)
library(pryr)
library(tinytex)
```

## Step 2: Set Today's date:
```{r set today, echo = FALSE}
today <- format(Sys.Date(), "%y%m%d")
today
```

## Step 3: Import 

Import the lists of changing genes from the Pollak paper generated in the DESeq2 analysis.
```{r, echo = FALSE}

# Obtain the Mek, Ras, and Akt Files:
upIn827_1 <- read.table(file = "../01_input/210405_1_upIn827.txt", header = TRUE)
upIn1502_2 <- read.table(file = "../01_input/210405_2_upIn1502.txt", header = TRUE)
upIn131_3 <- read.table(file = "../01_input/210405_3_upIn131.txt", header = TRUE)
upInG166_4 <- read.table(file = "../01_input/210405_4_upInG166.txt", header = TRUE)
upInG179_5 <- read.table(file = "../01_input/210405_5_upInG179.txt", header = TRUE)

downIn827_6 <- read.table(file = "../01_input/210405_6_downIn827.txt", header = TRUE)
downIn1502_7 <- read.table(file = "../01_input/210405_7_downIn1502.txt", header = TRUE)
downIn131_8 <- read.table(file = "../01_input/210405_8_downIn131.txt", header = TRUE)
downInG166_9 <- read.table(file = "../01_input/210405_9_downInG166.txt", header = TRUE)
downInG179_10 <- read.table(file = "../01_input/210405_10_downInG179.txt", header = TRUE)

# import the list of all genes:
all_genes <- read.table(file = "../01_input/210215_unionAllGenes_.txt", header = TRUE)

#dim(upIn827_1)
#dim(upIn1502_2)
#dim(upIn131_3)
#dim(upInG166_4)
#dim(upInG179_5)
```

## Step 4: Import the datasets from other published papers - 
```{r, echo = FALSE}
# Function - match and lookup
match_and_lookup_geneid <- function(published_genesetlist){
  all_genes[which(all_genes$external_gene_name %in% published_genesetlist),]$ensembl_gene_id
}

# Function - identify missing genes
identifyMissingGenes <- function(published_genesetlist){
  setdiff(published_genesetlist, all_genes$external_gene_name)
}

# Import Mek dataset
Mek18 <- read.table(file = "../01_input/MekEtAl18.txt", header = FALSE)
colnames(Mek18)[1] <- "genenames"
# get ensembl Gene IDs
mek18_ensbl_geneID <- match_and_lookup_geneid(Mek18$genenames)
#identifyMissingGenes(Mek18$genenames)
missing_ensembl <- c("ENSG00000107104", "ENSG00000103994")
mek18_ensbl_geneID <- c(mek18_ensbl_geneID, missing_ensembl)
#length(mek18_ensbl_geneID)
#length(Mek18$genenames)


# Import Pratilas dataset
Pratilas_table <- read.table(file = "../01_input/Pratilis_genenames.txt", header = FALSE)
colnames(Pratilas_table)[1] <- "genenames"
# get ensembl Gene IDs
Pratilas_ensbl_geneID <- match_and_lookup_geneid(Pratilas_table$genenames)
#identifyMissingGenes(Pratilas_table$genenames)
#missing
# "ALF"      "BXDC2"    "CD3EAP"   "FLJ10534" "HSPC111"  "IL8"      "KIR3DL2" 
# "LNK"      "PYCRL"
missing_ensmbl <- c("ENSG00000242441", "ENSG00000113460", "ENSG00000117877", "ENSG00000167721", "ENSG00000048162", "ENSG00000169429", "ENSG00000240403", "ENSG00000111252", "ENSG00000104524")
Pratilas_ensbl_geneID <- c(Pratilas_ensbl_geneID, missing_ensmbl)
#length(Pratilas_ensbl_geneID)
#length(Pratilas_table$genenames)

# Wagle et al -> 10 genes
vector_of_genes_wagle <- c("SPRY2", "SPRY4", "ETV4", "ETV5", "DUSP4", "DUSP6", "CCND1", "EPHA2", "EPHA4", "PHLDA1")
wagle_ensbl_geneID <- match_and_lookup_geneid(vector_of_genes_wagle)
#identifyMissingGenes(vector_of_genes_wagle)
# no missing genes

# Import Creighton dataset
creighton_table <- read.table(file = "../01_input/Creighton_list.txt", header = FALSE)
colnames(creighton_table)[1] <- "genenames"
creighton_ensbl_geneID <- match_and_lookup_geneid(creighton_table$genenames)
#identifyMissingGenes(creighton_table$genenames)
missing_ensmbl <- c("ENSG00000197451", "ENSG00000102225", "ENSG00000130202", "ENSG00000064601", "ENSG00000188229")
creighton_ensbl_geneID <- c(creighton_ensbl_geneID, missing_ensmbl)
#length(creighton_ensbl_geneID)
#length(creighton_table$genenames)


# Import Loboda dataset
lobodaUp_table <- read.table(file = "../01_input/Loboda_RASUp.txt", header = FALSE)
colnames(lobodaUp_table)[1] <- "genenames"
lobodaUp_ensbl_geneID <- match_and_lookup_geneid(lobodaUp_table$genenames)
#identifyMissingGenes(lobodaUp_table$genenames)
# "C19orf10" "C20orf42" "EREG"     "IL8"      "NPAL1"    "PBEF1"    "PTGS2"    "RPRC1"    "SERPINB5" "TNS4"    
# "TUBA1"    "VEGF"  
missing_ensmbl <- c("ENSG00000074842","ENSG00000101311", "ENSG00000124882", "ENSG00000169429", "ENSG00000163293", "ENSG00000105835", "ENSG00000073756", "ENSG00000116871",  "ENSG00000206075", "ENSG00000131746", "ENSG00000127824", "ENSG00000112715")
lobodaUp_ensbl_geneID <- c(lobodaUp_ensbl_geneID, missing_ensmbl)
#length(lobodaUp_ensbl_geneID)
#length(lobodaUp_table$genenames)
```

## Step 5: Writing functions 1 - 

Write a function called *make_contingency_table* that creates contingency tables for statistical analyses:

Write a function called *vector_of_p_values* that creates a vector of p-values generated from comparisons with the DESeq2 differentially expressed genes expressing the likeliness that a given gene signature set of genes are **enriched** amongst the list of genes.

Write a function called *vector_of_depletion_p_values* that creates a vector of p=values generated from comparisons with the DESeq2 differentially expressed genes expressing the likeliness that a given gene signature set of genes are **depleted** amongst the list of genes.

```{r, echo = FALSE}
make_contingency_table <- function(Published_list, RNAseqList, totalList) {
  intersectA <- length(intersect(Published_list, RNAseqList))
  NotFoundB <- length(Published_list) - intersectA
  MissedC <- length(RNAseqList) - intersectA
  NeitherD <- length(totalList) - sum(intersectA, NotFoundB, MissedC)
  
  table1 <- rbind(RNAseqChanging = c(intersectA, MissedC), RNAseqUnchanged = c(NotFoundB, NeitherD))
  colnames(table1) <- c("published", "Notpublished")
  return(as.table(table1))
}



# return a vector of P values assoicated with enrichment in each RNA-seq geneset
vector_of_p_values <- function(Published_list){
  p_value_vector <- c()
  table <- make_contingency_table(Published_list, upIn827_1$EnsembleGeneName, 
                                  all_genes$ensembl_gene_id)
  p_value_vector[1] <- fisher.test(table, alternative = "greater")$p.value

  table <- make_contingency_table(Published_list, upIn1502_2$EnsembleGeneName, 
                                  all_genes$ensembl_gene_id)
  p_value_vector[2] <- fisher.test(table, alternative = "greater")$p.value  
  
  table <- make_contingency_table(Published_list, upIn131_3$EnsembleGeneName, 
                                  all_genes$ensembl_gene_id)
  p_value_vector[3] <- fisher.test(table, alternative = "greater")$p.value   
  
  table <- make_contingency_table(Published_list, upInG166_4$EnsembleGeneName, 
                                  all_genes$ensembl_gene_id)
  p_value_vector[4] <- fisher.test(table, alternative = "greater")$p.value  
  
  table <- make_contingency_table(Published_list, upInG179_5$EnsembleGeneName, 
                                  all_genes$ensembl_gene_id)
  p_value_vector[5] <- fisher.test(table, alternative = "greater")$p.value   
  
  return(p_value_vector)
}


# return a vector of P values associated with depletion in each RNA-seq geneset
vector_of_depletion_p_values <- function(Published_list){
  p_value_vector <- c()
  table <- make_contingency_table(Published_list, downIn827_6$V1, all_genes$ensembl_gene_id)
  p_value_vector[1] <- fisher.test(table, alternative = "less")$p.value

  table <- make_contingency_table(Published_list, downIn1502_7$V1, all_genes$ensembl_gene_id)
  p_value_vector[2] <- fisher.test(table, alternative = "less")$p.value  
  
  table <- make_contingency_table(Published_list, downIn131_8$V1, all_genes$ensembl_gene_id)
  p_value_vector[3] <- fisher.test(table, alternative = "less")$p.value   

  table <- make_contingency_table(Published_list, downInG166_9$V1, all_genes$ensembl_gene_id)
  p_value_vector[4] <- fisher.test(table, alternative = "less")$p.value  
  
  table <- make_contingency_table(Published_list, downInG179_10$V1, all_genes$ensembl_gene_id)
  p_value_vector[5] <- fisher.test(table, alternative = "less")$p.value 
  return(p_value_vector)
}




```


Make contingency table creates contingency tables like so:

```{r}
exampletable <- make_contingency_table(lobodaUp_ensbl_geneID , upIn1502_2$EnsembleGeneName, all_genes$ensembl_gene_id)
exampletable

```

P values for enrichment are calculated from these contingency tables using Fisher exact test: 

```{r}
examplePvalue <- fisher.test(exampletable, alternative = "greater")$p.value
examplePvalue


```

## Step 6: Writing functions 2 - 

Write a function called *calculate_percent_enrichment* that calculates the over-representation of a published gene signature set of genes within the gene sets from GME RNA-seq analysis. The gene sets lists here are up-regulated genes published in Pollak et al. that we determined were over-expressed in each GME cell line.


Write a function called *calculate_fold_enrichment* that calcualets the fold enrichment between a published genes signature set and the genesets from our RNA-seq analysis.

```{r, echo = FALSE}

calculate_percent_enrichment <- function(vector_of_published_genes){
  percent <- c()
  
  table <- make_contingency_table(vector_of_published_genes, upIn827_1$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[11] <- sum(table[,1])/sum(table)*100
  
  percent[1] <- sum(table[1,1])/sum(table[1,])*100
  
  table <- make_contingency_table(vector_of_published_genes, upIn1502_2$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[2] <- sum(table[1,1])/sum(table[1,])*100
  
  table <- make_contingency_table(vector_of_published_genes, upIn131_3$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[3] <- sum(table[1,1])/sum(table[1,])*100
  
  table <- make_contingency_table(vector_of_published_genes, upInG166_4$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[4] <- sum(table[1,1])/sum(table[1,])*100
  
  table <- make_contingency_table(vector_of_published_genes, upInG179_5$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[5] <- sum(table[1,1])/sum(table[1,])*100

  table <- make_contingency_table(vector_of_published_genes, downIn827_6$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[6] <- sum(table[1,1])/sum(table[1,])*100
  
  table <- make_contingency_table(vector_of_published_genes, downIn1502_7$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[7] <- sum(table[1,1])/sum(table[1,])*100
  
  table <- make_contingency_table(vector_of_published_genes, downIn131_8$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[8] <- sum(table[1,1])/sum(table[1,])*100
  
  table <- make_contingency_table(vector_of_published_genes, downInG166_9$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[9] <- sum(table[1,1])/sum(table[1,])*100
  
  table <- make_contingency_table(vector_of_published_genes, downInG179_10$EnsembleGeneName, all_genes$ensembl_gene_id)
  percent[10] <- sum(table[1,1])/sum(table[1,])*100

  names(percent) <- c("UpIn827", "UpIn1502", "UpIn131", "UpInG166", "UpInG179", "DownIn827", "DownIn1502", "DownIn131", "DownInG166", "DownInG179", "Genomewide")

  return(percent)
}

#calculate_percent_enrichment(Pratilas_ensbl_geneID)

calculate_fold_enrichment <- function(vector_of_published_genes){
    percent <- c()
    foldEnrichment <- c()
    
    table <- make_contingency_table(vector_of_published_genes, upIn827_1$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[11] <- sum(table[,1])/sum(table)*100
    percent[1] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[1] <- percent[1] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, upIn1502_2$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[2] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[2] <- percent[2] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, upIn131_3$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[3] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[3] <- percent[3] / percent[11]
  
table <- make_contingency_table(vector_of_published_genes, upInG166_4$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[4] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[4] <- percent[4] / percent[11]
    
table <- make_contingency_table(vector_of_published_genes, upInG179_5$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[5] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[5] <- percent[5] / percent[11]

    table <- make_contingency_table(vector_of_published_genes, downIn827_6$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[6] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[6] <- percent[6] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downIn1502_7$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[7] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[7] <- percent[7] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downIn131_8$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[8] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[8] <- percent[8] / percent[11]
  
    table <- make_contingency_table(vector_of_published_genes, downInG166_9$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[9] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[9] <- percent[9] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downInG179_10$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[10] <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment[10] <- percent[10] / percent[11]
    names(foldEnrichment) <- c("UpIn827", "UpIn1502", "UpIn131", "UpInG166", "UpInG179", "DownIn827", "DownIn1502", "DownIn131", "DownInG166", "DownInG179")
  
    return(foldEnrichment)
}



calculate_fold_depletion <- function(vector_of_published_genes){
    percent <- c()
    foldDepletion <- c()
    
    table <- make_contingency_table(vector_of_published_genes, downIn827_6$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[11] <- sum(table[,1])/sum(table)*100
    percent[1] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[1] <- percent[1] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downIn1502_7$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[2] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[2] <- percent[2] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downIn131_8$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[3] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[3] <- percent[3] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downInG166_9$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[4] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[4] <- percent[4] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downInG179_10$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[5] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[5] <- percent[5] / percent[11]

    table <- make_contingency_table(vector_of_published_genes, downIn827_6$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[6] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[6] <- percent[6] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downIn1502_7$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[7] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[7] <- percent[7] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downIn131_8$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[8] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[8] <- percent[8] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downInG166_9$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[9] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[9] <- percent[9] / percent[11]
    
    table <- make_contingency_table(vector_of_published_genes, downInG179_10$EnsembleGeneName, all_genes$ensembl_gene_id)
    percent[10] <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion[10] <- percent[10] / percent[10]
  
    names(foldDepletion) <- c("UpIn827", "UpIn1502", "UpIn131", "UpInG166", "UpInG179", "DownIn827", "DownIn1502", "DownIn131", "DownInG166", "DownInG179")
  
    return(foldDepletion)
}

```


Percent enrichment is calculated as:$$\text{Percent enrichment}= \displaystyle \left(\frac{\text{set and list intersection}}{\text{GME list length}}\right)100\%$$



Fold enrichemnt is the percent calculation above divided by a similar calculation performed for the genome as a whole (instead of # in GME list)

Fold enrichment calculated as:$$\text{Fold enrichment}= \displaystyle \frac{\text{percent enrichment in GME}}{\text{percent enrichment genomewide}}$$

```{r}

exampletable
examplePercent <- sum(exampletable[1,1])/sum(exampletable[1,])*100
examplePercent

percentGenomewide <- sum(exampletable[,1])/sum(exampletable)*100
percentGenomewide
foldEnrichmentExample <- examplePercent/ percentGenomewide

foldEnrichmentExample
```

```

## Step 7: Gather statistics

```{r}
######################
# P-values
######################

# Generate an empty dataframe for the p-values calculating statistical significance of enrichment:
Pvalue_df <- data.frame(UpIn827 = numeric(),
                        UpIn1502 = numeric(),
                        UpIn131 = numeric(),
                        UpInG166 = numeric(),
                        UpInG179 = numeric(),
                        DownIn827 = numeric(),
                        DownIn1502 = numeric(),
                        DownIn131 = numeric(),
                        DownInG166 = numeric(),
                        DownInG179 = numeric())

# Save P-values of statistical enrichment to a dataframe called Pvalue_df
Pvalue_df[1,] <- vector_of_p_values(mek18_ensbl_geneID)
Pvalue_df[2,] <- vector_of_p_values(Pratilas_ensbl_geneID)
Pvalue_df[3,] <- vector_of_p_values(wagle_ensbl_geneID)
Pvalue_df[4,] <- vector_of_p_values(creighton_ensbl_geneID)
Pvalue_df[5,] <- vector_of_p_values(lobodaUp_ensbl_geneID)

rownames(Pvalue_df)[1] <- "Dry_et_al_mek_18"
rownames(Pvalue_df)[2] <- "Pratilas_et_al_mek_50"
rownames(Pvalue_df)[3] <- "Wagle_et_al_mek_10"
rownames(Pvalue_df)[4] <- "Creighton_et_al_akt_57"
rownames(Pvalue_df)[5] <- "Loboda_ras_101"

#Pvalue_df

######################
# Percent
######################

# Generate an empty dataframe for the percentages:
percent_df <- data.frame(UpIn827 = numeric(),
                        UpIn1502 = numeric(),
                        UpIn131 = numeric(),
                        UpInG166 = numeric(),
                        UpInG179 = numeric(),
                        DownIn827 = numeric(),
                        DownIn1502 = numeric(),
                        DownIn131 = numeric(),
                        DownInG166 = numeric(),
                        DownInG179 = numeric(),
                        Genomewide = numeric())

# Save percents to a dataframe called percent_df
percent_df[1,] <- calculate_percent_enrichment(mek18_ensbl_geneID)
percent_df[2,] <- calculate_percent_enrichment(Pratilas_ensbl_geneID)
percent_df[3,] <- calculate_percent_enrichment(wagle_ensbl_geneID)
percent_df[4,] <- calculate_percent_enrichment(creighton_ensbl_geneID)
percent_df[5,] <- calculate_percent_enrichment(lobodaUp_ensbl_geneID)

rownames(percent_df)[1] <- "Dry_et_al_mek_18"
rownames(percent_df)[2] <- "Pratilas_et_al_mek_50"
rownames(percent_df)[3] <- "Wagle_et_al_mek_10"
rownames(percent_df)[4] <- "Creighton_et_al_akt_57"
rownames(percent_df)[5] <- "Loboda_ras_101"

#percent_df

######################
# Fold Enrichment
######################

# Generate an empty dataframe for the fold enrichment:
df_enrich <- data.frame(UpIn827 = numeric(),
                        UpIn1502 = numeric(),
                        UpIn131 = numeric(),
                        UpInG166 = numeric(),
                        UpInG179 = numeric(),
                        DownIn827 = numeric(),
                        DownIn1502 = numeric(),
                        DownIn131 = numeric(),
                        DownInG166 = numeric(),
                        DownInG179 = numeric())

# Gather statistics: Fold enrichment - enrichment each list
df_enrich[1,] <- calculate_fold_enrichment(mek18_ensbl_geneID)
df_enrich[2,] <- calculate_fold_enrichment(Pratilas_ensbl_geneID)
df_enrich[3,] <- calculate_fold_enrichment(wagle_ensbl_geneID)
df_enrich[4,] <- calculate_fold_enrichment(creighton_ensbl_geneID)
df_enrich[5,] <- calculate_fold_enrichment(lobodaUp_ensbl_geneID)

rownames(df_enrich)[1] <- "Dry_et_al_mek_18"
rownames(df_enrich)[2] <- "Pratilas_et_al_mek_50"
rownames(df_enrich)[3] <- "Wagle_et_al_mek_10"
rownames(df_enrich)[4] <- "Creighton_et_al_akt_57"
rownames(df_enrich)[5] <- "Loboda_ras_101"

#df_enrich

######################
# Fold Depletion
######################

# Generate an empty dataframe for the fold depletion:
df_deplete <- data.frame(UpIn827 = numeric(),
                        UpIn1502 = numeric(),
                        UpIn131 = numeric(),
                        UpInG166 = numeric(),
                        UpInG179 = numeric(),
                        DownIn827 = numeric(),
                        DownIn1502 = numeric(),
                        DownIn131 = numeric(),
                        DownInG166 = numeric(),
                        DownInG179 = numeric())

# Gather statistics: Fold depletion - depletion in each list
df_deplete[1,] <- calculate_fold_depletion(mek18_ensbl_geneID)
df_deplete[2,] <- calculate_fold_depletion(Pratilas_ensbl_geneID)
df_deplete[3,] <- calculate_fold_depletion(wagle_ensbl_geneID)
df_deplete[4,] <- calculate_fold_depletion(creighton_ensbl_geneID)
df_deplete[5,] <- calculate_fold_depletion(lobodaUp_ensbl_geneID)

rownames(df_deplete)[1] <- "Dry_et_al_mek_18"
rownames(df_deplete)[2] <- "Pratilas_et_al_mek_50"
rownames(df_deplete)[3] <- "Wagle_et_al_mek_10"
rownames(df_deplete)[4] <- "Creighton_et_al_akt_57"
rownames(df_deplete)[5] <- "Loboda_ras_101"

#df_deplete

```

## Step 8: Barplots of enrichment/Depletion

```{r, echo = FALSE}

# Plot barplot of Dry et al, Mek
p_mek18 %<a-% {
  barplot(rev(as.matrix(percent_df[1,c(1:5,11)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Dry et al., 2010; 18 MEK genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[1,c(1:5,11)])))
  abline(v=as.numeric(percent_df[1,11]), col="red",lty=2)
}



p_mek18


# Plot of Pratilas et al., Mek
p_pratilas %<a-% {
  barplot(rev(as.matrix(percent_df[2,c(1:5,11)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Pratilas, MEK genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[2,c(1:5,11)])))
  abline(v=as.numeric(percent_df[2,11]), col="red",lty=2)
}

p_pratilas

# Plot of Wagle et al., Mek
p_wagle %<a-% {
  barplot(rev(as.matrix(percent_df[3,c(1:5,11)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Wagle, MEK genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[3,c(1:5,11)])))
  abline(v=as.numeric(percent_df[3,11]), col="red",lty=2)
}

p_wagle

# Plot of Wagle et al., Mek
p_creighton %<a-% {
  barplot(rev(as.matrix(percent_df[4,c(1:5,11)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Creighton et al, AKT genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[4,c(1:5,11)])))
  abline(v=as.numeric(percent_df[4,11]), col="red",lty=2)
}

p_creighton


p_loboda %<a-% {
  barplot(rev(as.matrix(percent_df[5,c(1:5,11)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Loboda et al, RAS genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[5,c(1:5,11)])))
  abline(v=as.numeric(percent_df[5,11]), col="red",lty=2)
}

p_loboda


```

## Step 9: Dot plot:

```{r, echo = FALSE}


# Subset the enrichment p-values to those I will use in a figure:
matrix_of_neg_log_Pvalues <- -log(Pvalue_df[c(4,2,1,3,5),1:5])

xlabelnames <- colnames(Pvalue_df[c(4,2,1,3,5),1:5])
ylabelnames <- rownames(Pvalue_df[c(4,2,1,3,5),1:5])


# Gather the fold change data 
df_enrich_reorder <- df_enrich[c(4,2,1,3,5),1:5]
melted_foldEnrich <- melt(df_enrich_reorder, value.name = "fold_enrich")

# Reshape the enrichment p-values into a long format dataset:
long_Pvalues <- as.matrix(matrix_of_neg_log_Pvalues) %>% 
  melt() %>%
  mutate(RNAseq_num = c(rep(1, 5), rep(2,5), rep(3,5), rep(4,5), rep(5,5))) %>%
  mutate(PubList = c(rep(c(1,2,3,4,5), 5))) %>%
  mutate(foldenrich = melted_foldEnrich$fold_enrich)

long_Pvalues$PubList <- as.factor(long_Pvalues$PubList)
colnames(long_Pvalues)[3] <- "neglogpvalue"

        
# Plot the data - enrichment
plot1 <- ggplot(long_Pvalues, aes(x=RNAseq_num, y=PubList, color=log(foldenrich))) + 
  geom_point(aes(size=neglogpvalue)) +
  scale_colour_gradient2(midpoint = 0, low = "navy", mid = "grey", high = "magenta", limits = c(-3, 3)) +
  scale_size(range = c(1, 20)) +
  scale_x_discrete(name = "Pollak et al., gene list", labels=xlabelnames, limits = factor(c(1, 2, 3, 4, 5))) +
  scale_y_discrete(name = "Published gene signature sets", labels=rev(ylabelnames), limits = c("5", "4", "3", "2", "1")) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), panel.grid.major = )


plot1

filename = paste("../03_output/", today, "_pvalue_enrich_dotplot.pdf", sep = "")
pdf(file = filename, width = 7.5, height = 4.5)
plot1
dev.off()

# Matrix of P-values un-transformed
print("Here is a matrix of P-values")
Pvalue_df

# Matrix of negative log P-values
print("Here is a matrix of negative log-transformed P-values")
matrix_of_neg_log_Pvalues

# Matrix of enrichment values
print("Here is a matrix of fold-enrichment values")
df_enrich_reorder


# Matrix of log transformed enrichment values
print("Here is a matrix of log-transformed fold-enrichment values")
log(df_enrich_reorder)

# Matrix of depletion values
print("Here is a matrix of fold-depletion values")
df_deplete


```



Session info:

```{r, echo = TRUE}
sessionInfo()
citation()
```

