---
title: "Analyze Glioblastoma RNA-seq Data"
author: "Romario Romain & Erin Osborne Nishimura"
date: "4/5/2021"
output: html_document
---

# Analysis: DESeq2 analysis

The purpose of this analysis is to analyze Pollak et al., 2017 paper that presented RNA-seq data on neuronal cells versus Glioblastoma cancer cells.

## Step 1: Install required libraries if not already present; Load libraries:

Install the following if not already installed:

  + DESeq2
  + RColorBrewer
  + pheatmap
  + ggplot2
  + BiocManager
  + GenomicFeatures
  + ensembldb
  + biomaRt
  + dplyr
  + tibble
  + tidyverse
  + gridExtra


```{r install packages, echo = FALSE}
# #Install bioconductor R version 3.6 or greater
#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install(version = "3.12")
# 
# ## When prompted to install (a/s) all or some of the dependent packages, move your cursor down to the console and type "a" for all
# 
#Install DESeq2:
# 
# library(BiocManager)
#  if (!requireNamespace("BiocManager", quietly = TRUE))
#          install.packages("BiocManager")
#   BiocManager::install("DESeq2")
#   BiocManager::install("biomaRt")
#   BiocManager::install("ensembldb")
#   BiocManager::install("GenomicFeatures")
#  BiocManager::install("apeglm")
# 
# 
# # #Install pretty heatmap - pheatmap: https://cran.r-project.org/web/packages/pheatmap/index.html
# install.packages("pheatmap")
# # 
# # #Install RColorBrewer: https://cran.r-project.org/web/packages/RColorBrewer/index.html
# install.packages("RColorBrewer")
# 
# install.packages("tidyverse")
# 
# BiocManager::valid()    
```

Load required packages.

```{r load packges, warning = FALSE, message = FALSE}

# Do this every time.
# Load packages:
library(DESeq2)
library(RColorBrewer)
library(pheatmap)
library(ggplot2)
library(BiocManager)
library(GenomicFeatures)
library(ensembldb)
library(biomaRt)
library(dplyr)
library(tibble)
library(tidyverse)
library(gridExtra)

```

Set date:

```{r, echo = FALSE}
today <- format(Sys.Date(), "%y%m%d")
today
```

## Step 2: Import the datasets

Import the counts dataset:

```{r read in data}

# If necessary:

# import the counts data - aligned to hg38
countsData <- read.table(file = "../01_input/counts_210403.txt", header = TRUE, row.names = 1, skip = 2) 

```

Import the metadata file:

```{r}
# Read in the metadata
metadata <- read.table(file = "../01_input/metadata_pollak_210402.txt", header = FALSE) # import the data
colnames(metadata) <- c("fasta1", "names", "celltype", "rep","cellOrigin")
metadata
```

## Step 3: Munge the data

Determine samples to analyze using the metadatafile

Append the metadata files to the countsData

```{r, echo = FALSE}
# The last names will be names for each sample. We can pull those names from metadata1:
#as.vector(metadata$names)
#as.vector(metadata$sample)

# Name countsData columns headers:
colnames(countsData) <- c("chr", "start", "stop", "strand", "length", as.vector(metadata$names))

```


Prepare the counts table:

```{r, echo = FALSE}

# Save just the subset as an object called cts:
cts <- as.matrix(countsData[,6:dim(countsData)[2]])

# Next we need to make an information called coltable. We can make this out of the metadata table.
# Reorganize the metadata table so the names1 column are now row headers
rownames(metadata)<- metadata$names

coldata <- metadata[,c("celltype", "rep", "names", "cellOrigin")]
coldata$celltype <- factor(coldata$celltype, levels = c("CB660", "827", "1502", "131", "G166", "G179"))
coldata$names <- as.factor(coldata$names)
coldata$rep <- as.factor(coldata$rep)
coldata$cellOrigin <- as.factor(coldata$cellOrigin)


```


One thing we need to do is explicitly check that the rownames of coldata exactly match the colnames of cts.

Do they match?

```{r}
#Check that the names match  --> Should be TRUE
all(rownames(coldata) == colnames(cts))
```


## Step 4: Perform DESeq2 analysis

Run the DESeq2 analysis on the dataset

```{r, echo = FALSE}

########## DESEQ2 Analysis ################
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ celltype)

```

Filter for present genes

Only keep genes that have 10 or greater reads associated with them.

```{r}
# Exclude genes with fewer than 10 reads
keep <- rowSums(counts(dds)) >=10 
dds <- dds[keep,]

# How many genes did we have before?
dim(cts)[1]

# How many genes do we have now?
dim(dds)[1]

```


Run the DESeq2 analysis

```{r}

# PERFORM DESEQ2 analysis:
####### This is where the magic happens! ########### 
# This will transform dds into a specialized object with many more fields filled in.
dds<- DESeq(dds)

# Organize the categories based on what makes sense:
dds$names<- factor(dds$names)

```

Exploratory data analysis

```{r}
#class(dds)
#str(dds)
#head(dds)
resultsNames(dds)
dim(dds)
plotDispEsts(dds)

```

What was the normalization scaling that was applied?

```{r, echo = FALSE}
# Size factor scaling
dds$sizeFactor

# Plot size factor scaling
par(mfrow = c(1,1))
barplot(as.vector(dds$sizeFactor), names.arg = as.vector(dds$names), ylim = c(0,3),
     main = "Normalization factoring for each sample", las = 2)


```

Excellent! We're seeing even representation across all samples.

## Step 5:  Append Ensemble identifiers to our dataset

First, import the gene names, either from EnsemblDB or from an input file:

```{r, echo = FALSE}
####### USE ENSEMBLE TO LOOKUP THE GENENAMES: - ONLY DO THIS THE FIRST TIME #######

# requires ensembldb
  # ensembl = useDataset("hsapiens_gene_ensembl",mart=useMart("ENSEMBL_MART_ENSEMBL"))
  # cts
  # ## Filter,values,and attributes at http://www.ensembl.org/biomart/martview
  # ctsnames = rownames(cts)
  # gene_names = getBM(mart = ensembl, filter='ensembl_gene_id', value=ctsnames, 
  #                   attributes=c('external_gene_name', 'ensembl_gene_id'))
  # filename <- paste("../01_input/", today, "_ensembleLookuptable.txt", sep = "")
  # write.table(gene_names, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
# Second time on...import the gene_names from the input folder
gene_names <-read.table(file = "../01_input/210116_ensembleLookuptable.txt", header = TRUE)
```

Write a script to append the gene names to the identifiers in a data.frame:

```{r, echo = FALSE}

# Functionify - identify the gene of interest
lookup_ensembl <- function(gene) {
  ensembl_name <- gene_names[which(gene_names$external_gene_name == gene),2]
  return(ensembl_name)
}

# Functionify - add ensemble gene names
add_external_name <- function(data_frame) {
  merged_data_frame <- merge(rownames_to_column(as.data.frame(data_frame)), gene_names, by.x = "rowname", by.y = "ensembl_gene_id", all.x = TRUE)
  merged_data_frame <- merged_data_frame %>% relocate(rowname, external_gene_name)
  colnames(merged_data_frame)[1] <- "EnsembleGeneName"
  return(merged_data_frame)
}


```


## Step 6: Export data tables. 

Some data tables to export and include in the study as supplemental data tables.

```{r, echo = TRUE}

# ###### RAW COUNTS ##########
# 
# # create raw counts tables for supplemental figures
raw_counts <- counts(dds, normalized = FALSE)
merged_raw_counts <- add_external_name(raw_counts)
# 
# ###### NORMALIZED COUNTS ##########
# 
# # create normalized counts tables for supplemental figures
 norm_counts <- counts(dds, normalized = TRUE)
 merged_norm_counts <- add_external_name(norm_counts)
# 
# ###### R-STABILIZED LOG COUNTS ##########
# rdl_all <- as.data.frame(assay(rlog(dds, blind=FALSE)))
# merged_rdl <- add_external_name(rdl_all)
# 
# ####### save raw counts file for supplemental datatable ######
# filename = paste("../03_output/", today, "_raw_counts.txt", sep = "")
# write.table(merged_raw_counts, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)
# 
# ####### save normalized raw counts file for supplemental datatable ######
# filename = paste("../03_output/", today, "_norm_counts.txt", sep = "")
# write.table(merged_norm_counts, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)
# 
# ####### save normalized raw counts file for supplemental datatable ######
# filename = paste("../03_output/", today, "_RStabilized_log_counts.txt", sep = "")
# write.table(merged_rdl, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)

```

## Step 7: Plot a distance matrix

```{r Sample distance matrix (shows sample similarity), echo = FALSE}
#Take r-stabilized log transformations of all the normalized count data. This will help with the problem that the data is noisy and it will help with the problem that the data is spread across a wide range of values.
rld <- rlog(dds, blind=FALSE)  #Take the r-stabilized log transformed data:

# Calculate the distances between each sample
sampleDists <- dist(t(assay(rld))) # calculate distance matrices:
sampleDistMatrix <- as.matrix(sampleDists) #convert from data.frame -> matrix
rownames(sampleDistMatrix) <- colnames(rld) # Add some labels
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255) #Pick some pretty colors

# Draw the heatmap
par(mfrow=c(1,1))
dist_matrix <- pheatmap::pheatmap(sampleDistMatrix,
                        clustering_distance_rows=sampleDists,
                        clustering_distance_cols=sampleDists,
                        col=colors, width = 4, height = 4,
                        cluster_rows = TRUE) # Plot the heatmap

figure_name_matrix <- paste("../03_output/", today, "_distanceMatrixPlot.pdf", sep = "")

head(cts)
dist_matrix

# save distance matrix figure
pdf(figure_name_matrix, height = 6, width = 7)
  par(mfrow=c(1,1))
  dist_matrix
  dev.off() 
```
Excellent separation between all conditions.


## Step 8: Generate a PCA plot:

```{r, echo = FALSE}

# PCA plot
PCA <- plotPCA(rld, intgroup=c("celltype"))
PCA <- PCA + theme(aspect.ratio=1)
PCA

# Save it to a .pdf
figure_PCA <- paste("../03_output/", today, "_PCAPlot.pdf", sep = "")
pdf(figure_PCA, height = 6, width = 5)
  PCA 
  dev.off() 

```

## Step 9: Plot some key, individual genes

Write the function plot_ensembl and gather genes:

Here are plots of some "Up in Glioblastoma" genes (from Jake H)

```{r individual gene plots, echo = FALSE} 

# Functionify - plotting
plot_ensembl <- function(gene) {
  plotcounts <- plotCounts(dds, gene=which(rownames(norm_counts) == gene),
                          intgroup = c("celltype"),
                          cex.main=2, cex=1.5, xlab= "cell type", returnData = T)
  common_name <- gene_names[which(gene_names$ensembl_gene_id == gene),1]
  label = paste(common_name, "normalized expression", sep = " ")
  
  cbPalette <- c("#77A540", "#4B79BD", "#E8BB20", "#F47C20", "#A84B44", "#0072B2", "#D55E00", "#CC79A7")

  p <- ggplot (plotcounts,aes(x=celltype,y=count)) +
    geom_boxplot(aes(fill=celltype))+
    geom_jitter(width = 0.25) +
    scale_fill_manual(values=cbPalette) +
    scale_y_log10() +
    ggtitle(label) +
    theme(text=element_text(family="Arial")) +
    theme_bw() +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
    theme(axis.text.x = element_text(angle = 90))
  return(p)
}

# Functionify - plotting
plot_ensembl_with_ensembl_name <- function(gene) {
  plotcounts <- plotCounts(dds, gene=which(rownames(norm_counts)== gene),
                          intgroup = c("sample"),
                          cex.main=2, cex=1.5, xlab= "Sample", returnData = T)
  #common_name <- gene_names[which(gene_names$ensembl_gene_id == gene),1]
  label = paste(gene, "normalized expression", sep = " ")
  
  cbPalette <- c("#77A540", "#4B79BD", "#E8BB20", "#F47C20", "#A84B44", "#0072B2", "#D55E00", "#CC79A7")

  p <- ggplot (plotcounts,aes(x=celltype,y=count)) +
    geom_boxplot(aes(fill=celltype))+
    geom_jitter(width = 0.25) +
    scale_fill_manual(values=cbPalette) +
    scale_y_log10() +
    ggtitle(label) +
    theme(text=element_text(family="Arial")) +
    theme_bw() +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
    theme(axis.text.x = element_text(angle = 90))
  return(p)
}

#Pull out the transgenes - Glioblastoma - linked hg38 aligned

vector_of_genes <- sort(c("COL3A1", "TOP2A", "CRISPLD1", "RRM2", "COL1A2", "FCGBP", "CDCA7L",
                          "SMC4", "TMEM45A","PTX3"))
vector_of_ensembl <- sapply(vector_of_genes, lookup_ensembl, simplify = TRUE)


# Make two lists of plots
plist <- lapply(vector_of_ensembl, plot_ensembl)

# Save individual plots
p1 <- plist[[1]]
p2 <- plist[[2]]
p3 <- plist[[3]]
p4 <- plist[[4]]
p5 <- plist[[5]]
p6 <- plist[[6]]
p7 <- plist[[7]]
p8 <- plist[[8]]
p9 <- plist[[9]]
p10 <- plist[[10]]

# Print plots in .html file
grid.arrange(p1, p2, nrow = 1)
grid.arrange(p3, p4, nrow = 1)
grid.arrange(p5, p6, nrow = 1)
grid.arrange(p7, p8, nrow = 1)
grid.arrange(p9, p10, nrow = 1)


# Print plots as .pdf files
figure_plotcounts1 <- paste("../03_output/", today, "_PlotCounts1.pdf", sep = "")
pdf(file = figure_plotcounts1, width = 8, height = 12 )
  grid.arrange(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10, nrow = 4)
  dev.off()
```


Here are some general cancer-related RAS/MEK/AKT genes:

```{r, echo = FALSE}

# Pull out genes - cancer RAS/MEK/AKT linked
vector_of_genes <- sort(c("CCND1", "CDK4", "TERT", "TP53", "AKT1", "MAP2K1", "HRAS"))

vector_of_ensembl <- sapply(vector_of_genes, lookup_ensembl, simplify = TRUE)

# Make two lists of plots
plist <- lapply(vector_of_ensembl, plot_ensembl)

# Save individual plots
p1 <- plist[[1]]
p2 <- plist[[2]]
p3 <- plist[[3]]
p4 <- plist[[4]]
p5 <- plist[[5]]
p6 <- plist[[6]]
p7 <- plist[[7]]

# Print plots in .html file
grid.arrange(p1, p2, nrow = 1)
grid.arrange(p3, p4, nrow = 1)
grid.arrange(p5, p6, nrow = 1)
grid.arrange(p7, p7, nrow = 1)

# Print plots as .pdf files
figure_plotcounts1 <- paste("../03_output/", today, "_PlotCounts2.pdf", sep = "")
pdf(file = figure_plotcounts1, width = 8, height = 12 )
  grid.arrange(p1, p2, p3, p4, p5, p6, p7, nrow = 4)
  dev.off()
  

```

## Step 10: Identify pairwise differentially expressed genes

Write function calc_pairwise_results to calculate pairwise results tables

Calculate all neuronal v. GBM comparisons

```{r differentially expressed genes}
# Write key functions to calculate Threshold values

calc_pairwise_results <- function(cond1, cond2, lfc){
  res <- results(dds, contrast=c("celltype", cond1, cond2), 
                 alpha = 0.01, lfcThreshold = lfc, cooksCutoff = FALSE)
  res <- res[order(res$padj, na.last = TRUE),]
  return(res)
}

# Perform pairwise comparisons against the neuronal type: CB660
# lfc = 1; default padj = 0.01
#CB660 827 1502 131 G166 G179
res_CB660_vs_827 <- calc_pairwise_results("827","CB660", 2)
res_CB660_vs_1502 <- calc_pairwise_results("1502", "CB660", 2)
res_CB660_vs_131 <- calc_pairwise_results("131", "CB660", 2)
res_CB660_vs_G166 <- calc_pairwise_results("G166", "CB660", 2)
res_CB660_vs_G179 <- calc_pairwise_results("G179", "CB660", 2)

 
```

Report the summaries of the res-plots:

```{r, echo = TRUE}

# Report summaries
summary(res_CB660_vs_827)
summary(res_CB660_vs_1502)
summary(res_CB660_vs_131)
summary(res_CB660_vs_G166)
summary(res_CB660_vs_G179)
```


## Step 11: Plot the MA-Plots

MA plots of pairwise comparisons

```{r, echo = FALSE}

# Plot MA-plots
par(mfrow=c(2,3))
plotMA(res_CB660_vs_827, main="CB660 v. 827", ylim=c(-20,20))
plotMA(res_CB660_vs_1502, main="CB660 v. 1502", ylim=c(-20,20))
plotMA(res_CB660_vs_131, main="CB660 v. 131", ylim=c(-20,20))
plotMA(res_CB660_vs_G166, main="CB660 v. G166", ylim=c(-20,20))
plotMA(res_CB660_vs_G179, main="CB660 v. G179", ylim=c(-20,20))

#  Step4 -> click here to see what you got!
#idx <- identify(res_ARPE19_vs_Aktmyr$baseMean, res_ARPE19_vs_Aktmyr$log2FoldChange)
#rownames(res_ARPE19_vs_Aktmyr)[idx]

```

Save MA plots to .pdf's:

```{r}
# Printout MA-Plots
filename = paste("../03_output/", today, "_MAplots.pdf", sep = "")
pdf(file = filename, height = 6, width = 6)
  par(mfrow=c(2,3))
  plotMA(res_CB660_vs_827, main="CB660 v. 827", ylim=c(-20,20))
  plotMA(res_CB660_vs_1502, main="CB660 v. 1502", ylim=c(-20,20))
  plotMA(res_CB660_vs_131, main="CB660 v. 131", ylim=c(-20,20))
  plotMA(res_CB660_vs_G166, main="CB660 v. G166", ylim=c(-20,20))
  plotMA(res_CB660_vs_G179, main="CB660 v. G179", ylim=c(-20,20))
  dev.off()
```

## Step 12: Generate and export gene lists

```{r}
# find the up_in_X and down_in_X genesets:
# lfc = 0.2; alpha = 0.1

# Up in 
Up_in_827 <- subset(res_CB660_vs_827, padj < 0.01 & log2FoldChange > 2)
Up_in_1502 <- subset(res_CB660_vs_1502, padj < 0.01 & log2FoldChange > 2)
Up_in_131 <- subset(res_CB660_vs_131, padj < 0.01 & log2FoldChange > 2)
Up_in_G166 <- subset(res_CB660_vs_G166, padj < 0.01 & log2FoldChange > 2)
Up_in_G179 <- subset(res_CB660_vs_G179, padj < 0.01 & log2FoldChange > 2)

# Down in 
Down_in_827 <- subset(res_CB660_vs_827, padj < 0.01 & log2FoldChange < -2)
Down_in_1502 <- subset(res_CB660_vs_1502, padj < 0.01 & log2FoldChange < -2)
Down_in_131 <- subset(res_CB660_vs_131, padj < 0.01 & log2FoldChange < -2)
Down_in_G166 <- subset(res_CB660_vs_G166, padj < 0.01 & log2FoldChange < -2)
Down_in_G179 <- subset(res_CB660_vs_G179, padj < 0.01 & log2FoldChange < -2)

# Add external names
merged_up_in_827 <- add_external_name(Up_in_827)
merged_up_in_1502 <- add_external_name(Up_in_1502)
merged_up_in_131 <- add_external_name(Up_in_131)
merged_up_in_G166 <- add_external_name(Up_in_G166)
merged_up_in_G179 <- add_external_name(Up_in_G179)

merged_down_in_827 <- add_external_name(Down_in_827)
merged_down_in_1502 <- add_external_name(Down_in_1502)
merged_down_in_131 <- add_external_name(Down_in_131)
merged_down_in_G166 <- add_external_name(Down_in_G166)
merged_down_in_G179 <- add_external_name(Down_in_G179)

# Write files - up lists:
filename1 = paste("../03_output/", today, "_1_upIn827.txt", sep = "")
write.table(file = filename1, merged_up_in_827, sep = "\t", quote = FALSE, row.names = FALSE)

filename1 = paste("../03_output/", today, "_2_upIn1502.txt", sep = "")
write.table(file = filename1, merged_up_in_1502, sep = "\t", quote = FALSE, row.names = FALSE)

filename1 = paste("../03_output/", today, "_3_upIn131.txt", sep = "")
write.table(file = filename1, merged_up_in_131, sep = "\t", quote = FALSE, row.names = FALSE)

filename1 = paste("../03_output/", today, "_4_upInG166.txt", sep = "")
write.table(file = filename1, merged_up_in_G166, sep = "\t", quote = FALSE, row.names = FALSE)

filename1 = paste("../03_output/", today, "_5_upInG179.txt", sep = "")
write.table(file = filename1, merged_up_in_G179, sep = "\t", quote = FALSE, row.names = FALSE)


# Write files - down lists:
filename1 = paste("../03_output/", today, "_6_upIn827.txt", sep = "")
write.table(file = filename1, merged_down_in_827, sep = "\t", quote = FALSE, row.names = FALSE)

filename1 = paste("../03_output/", today, "_7_downIn1502.txt", sep = "")
write.table(file = filename1, merged_down_in_1502, sep = "\t", quote = FALSE, row.names = FALSE)

filename1 = paste("../03_output/", today, "_8_downIn131.txt", sep = "")
write.table(file = filename1, merged_down_in_131, sep = "\t", quote = FALSE, row.names = FALSE)

filename1 = paste("../03_output/", today, "_9_downInG166.txt", sep = "")
write.table(file = filename1, merged_down_in_G166, sep = "\t", quote = FALSE, row.names = FALSE)

filename1 = paste("../03_output/", today, "_10_downInG179.txt", sep = "")
write.table(file = filename1, merged_down_in_G179, sep = "\t", quote = FALSE, row.names = FALSE)



```

## Step 14: Export the results tables and the DESeq object as files

```{r, echo = FALSE}


# Add gene names and sort in ascending order by adjusted p-value
df_res_CB660_vs_827 <- add_external_name(res_CB660_vs_827) %>% arrange(padj)
df_res_CB660_vs_1502 <- add_external_name(res_CB660_vs_1502) %>% arrange(padj)
df_res_CB660_vs_131 <- add_external_name(res_CB660_vs_131) %>% arrange(padj)
df_res_CB660_vs_G166 <- add_external_name(res_CB660_vs_G166) %>% arrange(padj)
df_res_CB660_vs_G179 <- add_external_name(res_CB660_vs_G179) %>% arrange(padj)

```

```{r}
# Write to .txt files
filename = paste("../03_output/", today, "_results_CB660_vs_827.txt", sep = "")
write.table(df_res_CB660_vs_827, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)

filename = paste("../03_output/", today, "_resultsTable_CB660_vs_1502.txt", sep = "")
write.table(df_res_CB660_vs_1502, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)

filename = paste("../03_output/", today, "_resultsTable_CB660_vs_131.txt", sep = "")
write.table(df_res_CB660_vs_131, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)

filename = paste("../03_output/", today, "_resultsTable_CB660_vs_G166.txt", sep = "")
write.table(df_res_CB660_vs_G166, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)

filename = paste("../03_output/", today, "_resultsTable_CB660_vs_G179.txt", sep = "")
write.table(df_res_CB660_vs_G179, file = filename, sep = "\t", quote = FALSE, row.names = FALSE)

filename_dds <- paste("../03_output/", today, "_dds_object.rds", sep = "")
saveRDS(dds, file = filename_dds)

```

## Step 15: Export session info

```{r, echo = TRUE}
sessionInfo()

citation()
```

