---
title: "210517_HeatmapsOfRelevantGenes_CohenSharir"
author: "Erin Osborne Nishimura"
date: "5/17/2021"
output: html_document
---

# Analysis: Heatmaps of key gene signature sets

The purpose of this analysis is to assess how gene signature sets are changing in the Cohen-Sharir, 2021 datasets. Previous publications have reported the transcriptional profiles associated with different types of pathway activation.

The Cohen-Sharir Dataset is here:
  
  * Cohen-Sharir Y, McFarland JM, Abdusamad M, Marquis C, Bernhard SV, Kazachkova M, Tang H, Ippolito MR, Laue K, Zerbib J, Malaby HLH, Jones A, Stautmeister LM, Bockaj I, Wardenaar R, Lyons N, Nagaraja A, Bass AJ, Spierings DCJ, Foijer F, Beroukhim R, Santaguida S, Golub TR, Stumpff J, Storchová Z, Ben-David U. [Aneuploidy renders cancer cells vulnerable to mitotic checkpoint inhibition](https://doi: 10.1038/s41586-020-03114-6) _Nature_ **2021** Feb;590(7846):486-491.

For this analysis, we used the following publications:

  * Creighton CJ. [A gene transcription signature of the Akt/mTOR pathway in clinical breast tumors](https://doi.org/10.1038/sj.onc.1210245). _Oncogene_ **2007**;26:4648–55.

  * Dry JR, Pavey S, Pratilas CA, Harbron C, Runswick S, Hodgson D, et al. [Transcriptional Pathway Signatures Predict MEK Addiction and Response to Selumetinib (AZD6244)](https://doi.org/10.1158/0008-5472.can-09-1577). _Cancer Res_ **2010**;70:2264–73.

  * Loboda A, Nebozhyn M, Klinghoffer R, Frazier J, Chastain M, Arthur W, et al. [A gene expression signature of RAS pathway dependence predicts response to PI3K and RAS pathway inhibitors and expands the population of RAS pathway activated tumors.](https://doi.org/10.1186/1755-8794-3-26) _Bmc Med Genomics_ **2010**;3:26.

  * Pratilas CA, Taylor BS, Ye Q, Viale A, Sander C, Solit DB, et al. [V600EBRAF is associated with disabled feedback inhibition of RAF–MEK signaling and elevated transcriptional output of the pathway.](https://doi.org/10.1073/pnas.0900780106) _Proc National Acad Sci_ **2009**;106:4519–24. 

  * Wagle M-C, Kirouac D, Klijn C, Liu B, Mahajan S, Junttila M, et al. [A transcriptional MAPK Pathway Activity Score (MPAS) is a clinically relevant biomarker in multiple cancer types](https://doi.org/10.1038/s41698-018-0051-4). _Npj Precis Oncol_ **2018**;2:7.

```{r, echo = FALSE}

# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install(version = "3.12")
# 
# library(BiocManager)
# BiocManager::install("ComplexHeatmap")
# I need to download X11 - XQuartz on Mac to use ComplexHeatmap


#if (!requireNamespace("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install(version = "3.12")

#BiocManager::install("clusterProfiler")


```



## Step 1: Load packages

```{r load packges, warning = FALSE, message = FALSE}

library(RColorBrewer)
library(ggplot2)
library(dplyr)
library(pheatmap)
library(reshape2)
library(pryr)
```

## Step 2: Set Today's date:

```{r set today, echo = FALSE}
today <- format(Sys.Date(), "%y%m%d")
today
```

## Step 3: Import the data from CohenSharir - 


```{r Importing}
# This was an output from: 210128_DESeq_analyze_EONversion
cohen_sharir <- read.table(file = "../01_input/41586_2020_3114_MOESM6_ESM.txt", header = TRUE) 

UpInCohenSharir <- rownames(cohen_sharir %>%
    filter(q_value < 0.05) %>%
      filter(Effect_size > 0 ))
length(UpInCohenSharir)

print("How many genes are over-expressed in the Cohen-Sharir dataset?")
length(UpInCohenSharir)

DownInCohenSharir <- rownames(cohen_sharir %>%
    filter(q_value < 0.05) %>%
      filter(Effect_size < 0 ))
length(DownInCohenSharir)
#DownInCohenSharir

```


## Step 4: Import the datasets from other published papers - 

```{r}

# Import Mek dataset
Mek18 <- read.table(file = "../01_input/MekEtAl18.txt", header = FALSE)
colnames(Mek18)[1] <- "genenames"

# Import Pratilas dataset
Pratilas_table <- read.table(file = "../01_input/Pratilis_genenames.txt", header = FALSE)
colnames(Pratilas_table)[1] <- "genenames"

# Wagle et al -> 10 genes
vector_of_genes_wagle <- c("SPRY2", "SPRY4", "ETV4", "ETV5", "DUSP4", "DUSP6", "CCND1", "EPHA2", "EPHA4", "PHLDA1")

# Import Creighton dataset
creighton_table <- read.table(file = "../01_input/Creighton_list.txt", header = FALSE)
colnames(creighton_table)[1] <- "genenames"

# Import Loboda dataset
lobodaUp_table <- read.table(file = "../01_input/Loboda_RASUp.txt", header = FALSE)
colnames(lobodaUp_table)[1] <- "genenames"

```

## Step 5: Write some functions:

Write a function called *make_contingency_table* that creates contingency tables for statistical analyses:

Write a function called *vector_of_p_values* that creates a vector of p-values generated from comparisons with the DESeq2 differentially expressed genes expressing the likeliness that a given gene signature set of genes are **enriched** amongst the list of genes.

Write a function called *vector_of_depletion_p_values* that creates a vector of p=values generated from comparisons with the DESeq2 differentially expressed genes expressing the likeliness that a given gene signature set of genes are **depleted** amongst the list of genes.

```{r, echo = FALSE}
make_contingency_table <- function(Published_list, RNAseqList, totalList) {
  Missing_genes <- setdiff(Published_list, totalList)
  message <- paste("There are ", Missing_genes, " missing genes.")
  #print(message)
  intersectA <- length(intersect(Published_list, RNAseqList))
  NotFoundB <- length(Published_list) - intersectA
  MissedC <- length(RNAseqList) - intersectA
  NeitherD <- length(totalList) - sum(intersectA, NotFoundB, MissedC)
  table1 <- rbind(RNAseqChanging = c(intersectA, MissedC), RNAseqUnchanged = c(NotFoundB, NeitherD))
  colnames(table1) <- c("published", "Notpublished")
  return(as.table(table1))
}


# return a vector of P values assoicated with enrichment in each RNA-seq geneset
vector_of_p_values <- function(Published_list){
  p_value_vector <- c()
  
  table <- make_contingency_table(Published_list, UpInCohenSharir, rownames(cohen_sharir))
  p_value_vector[1] <- fisher.test(table, alternative = "greater")$p.value

  table <- make_contingency_table(Published_list, DownInCohenSharir, rownames(cohen_sharir))
  p_value_vector[2] <- fisher.test(table, alternative = "greater")$p.value  
  
  return(p_value_vector)
}

vector_of_p_values(Mek18$genenames)

# return a vector of P values associated with depletion in each RNA-seq geneset
vector_of_depletion_p_values <- function(Published_list){
  p_value_vector <- c()
  
  table <- make_contingency_table(Published_list, UpInCohenSharir, rownames(cohen_sharir))
  p_value_vector[1] <- fisher.test(table, alternative = "less")$p.value

  table <- make_contingency_table(Published_list, DownInCohenSharir, rownames(cohen_sharir))
  p_value_vector[2] <- fisher.test(table, alternative = "less")$p.value  

  return(p_value_vector)
}


```
## Step 6: Write some functions:

Write a function called *calculate_percent_enrichment* that calculates the over-representation of a published gene signature set of genes within the gene sets from GME RNA-seq analysis. The gene sets lists here are up-regulated genes published in Cohen-Sharir et al. that we determined were over-expressed in these cells.


Write a function called *calculate_fold_enrichment* that calcualets the fold enrichment between a published genes signature set and the genesets from our RNA-seq analysis.

```{r, echo = FALSE}


calculate_percent_enrichment <- function(vector_of_published_genes){
  percent <- c()
  
  # Calculate % Published genes in UpInCohen Set
  table1 <- make_contingency_table(vector_of_published_genes, UpInCohenSharir, rownames(cohen_sharir))
  percent[1] <- sum(table1[1,1])/sum(table1[1,])*100

  # Calculate % Published genes in DownInCohen Set
  table2 <- make_contingency_table(vector_of_published_genes, DownInCohenSharir, rownames(cohen_sharir))
  percent[2] <- sum(table2[1,1])/sum(table2[1,])*100

  # Calculate % Published genes in All genes
  percent[3] <- sum(table2[,1])/sum(table2)*100 

  names(percent) <- c("PercentUpInCohenSharir", "PercentDownInCohenSharir", "PercentInAll")

  return(percent)
}
  



calculate_fold_enrichment <- function(vector_of_published_genes){
    # Calculate fold enrichment
    table <- make_contingency_table(vector_of_published_genes, UpInCohenSharir, rownames(cohen_sharir))
    percent_all <- sum(table[,1])/sum(table)*100
    percent_up <- sum(table[1,1])/sum(table[1,])*100
    foldEnrichment <- percent_up / percent_all
    
    # Label it
    names(foldEnrichment) <- c("UpInCS")
    
    # Return it
    return(foldEnrichment)
}

calculate_fold_depletion <- function(vector_of_published_genes){
    # Calculate fold depletion    
    table <- make_contingency_table(vector_of_published_genes, DownInCohenSharir, rownames(cohen_sharir))
    percent_all <- sum(table[,1])/sum(table)*100
    percent_down <- sum(table[1,1])/sum(table[1,])*100
    foldDepletion <- percent_down / percent_all
    
    # Label it
    names(foldDepletion) <- c("DownInCS")
    
    # Return it
    return(foldDepletion)
}


```


## Step 7: Gather statistics:

```{r, echo = FALSE}

# Gather statistics: Percent representation
# What percent of the Up in CS or Down in CS lists are represented by the published lists?

######################
# P-values
######################

# Generate an empty dataframe for the p-values calculating statistical significance of enrichment as compared to genomewide representation:
Pvalue_df <- data.frame(UpInCohenSharir = numeric(),
                        DownInCohenSharir = numeric())

# Save P-values of statistical enrichment to a dataframe called Pvalue_df
Pvalue_df[1,] <- vector_of_p_values(Mek18$genenames)
Pvalue_df[2,] <- vector_of_p_values(Pratilas_table$genenames)
Pvalue_df[3,] <- vector_of_p_values(vector_of_genes_wagle)
Pvalue_df[4,] <- vector_of_p_values(creighton_table$genenames)
Pvalue_df[5,] <- vector_of_p_values(lobodaUp_table$genenames)

rownames(Pvalue_df)[1] <- "Dry_et_al_mek_18"
rownames(Pvalue_df)[2] <- "Pratilas_et_al_mek_50"
rownames(Pvalue_df)[3] <- "Wagle_et_al_mek_10"
rownames(Pvalue_df)[4] <- "Creighton_et_al_akt_57"
rownames(Pvalue_df)[5] <- "Loboda_ras_101"

print("P-values of the significance of enrichment")
Pvalue_df

######################
# Percent
######################

# Generate an empty dataframe for the percentages:
percent_df <- data.frame(UpInCohenSharir = numeric(),
                        DownInCohenSharir = numeric(),
                        Genomewide = numeric())

# Save percents to a dataframe called percent_df
percent_df[1,] <- calculate_percent_enrichment(Mek18$genenames)
percent_df[2,] <- calculate_percent_enrichment(Pratilas_table$genenames)
percent_df[3,] <- calculate_percent_enrichment(vector_of_genes_wagle)
percent_df[4,] <- calculate_percent_enrichment(creighton_table$genenames)
percent_df[5,] <- calculate_percent_enrichment(lobodaUp_table$genenames)

rownames(percent_df)[1] <- "Dry_et_al_mek_18"
rownames(percent_df)[2] <- "Pratilas_et_al_mek_50"
rownames(percent_df)[3] <- "Wagle_et_al_mek_10"
rownames(percent_df)[4] <- "Creighton_et_al_akt_57"
rownames(percent_df)[5] <- "Loboda_ras_101"

print("Actual percents of of enrichment")
percent_df

######################
# Fold Enrichment
######################

# Generate an empty dataframe for the fold enrichment:
df_enrich <- data.frame(UpInCohenSharir = numeric())

# Gather statistics: Fold enrichment - enrichment each list
df_enrich[1,] <- calculate_fold_enrichment(Mek18$genenames)
df_enrich[2,] <- calculate_fold_enrichment(Pratilas_table$genenames)
df_enrich[3,] <- calculate_fold_enrichment(vector_of_genes_wagle)
df_enrich[4,] <- calculate_fold_enrichment(creighton_table$genenames)
df_enrich[5,] <- calculate_fold_enrichment(lobodaUp_table$genenames)

rownames(df_enrich)[1] <- "Dry_et_al_mek_18"
rownames(df_enrich)[2] <- "Pratilas_et_al_mek_50"
rownames(df_enrich)[3] <- "Wagle_et_al_mek_10"
rownames(df_enrich)[4] <- "Creighton_et_al_akt_57"
rownames(df_enrich)[5] <- "Loboda_ras_101"

print("Fold change of of enrichment")
df_enrich

######################
# Fold Depletion
######################

# Generate an empty dataframe for the fold depletion:
df_deplete <- data.frame(DownInCohenSharir = numeric())


# Gather statistics: Fold depletion - depletion in each list
df_deplete[1,] <- calculate_fold_depletion(Mek18$genenames)
df_deplete[2,] <- calculate_fold_depletion(Pratilas_table$genenames)
df_deplete[3,] <- calculate_fold_depletion(vector_of_genes_wagle)
df_deplete[4,] <- calculate_fold_depletion(creighton_table$genenames)
df_deplete[5,] <- calculate_fold_depletion(lobodaUp_table$genenames)

rownames(df_deplete)[1] <- "Dry_et_al_mek_18"
rownames(df_deplete)[2] <- "Pratilas_et_al_mek_50"
rownames(df_deplete)[3] <- "Wagle_et_al_mek_10"
rownames(df_deplete)[4] <- "Creighton_et_al_akt_57"
rownames(df_deplete)[5] <- "Loboda_ras_101"

print("Fold depletion")
df_deplete



```

## Step 8: Barplots of enrichment/Depletion

Plot some barplots to illustrate the percentage of each Cohen-Sharir up-regulated genes and their representation in each published gene signature set.

```{r, echo = FALSE}

# Plot barplot of Dry et al, Mek
p_mek18 %<a-% {
  barplot(rev(as.matrix(percent_df[1,c(1:3)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Dry et al., 2010; 18 MEK genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[1,c(1:3)])))
  abline(v=as.numeric(percent_df[1,3]), col="red",lty=2)
}

p_mek18

# Plot of Pratilas et al., Mek
p_pratilas %<a-% {
  barplot(rev(as.matrix(percent_df[2,c(1:3)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Pratilas, MEK genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[2,c(1:3)])))
  abline(v=as.numeric(percent_df[2,3]), col="red",lty=2)
}

p_pratilas

# Plot of Wagle et al., Mek
p_wagle %<a-% {
  barplot(rev(as.matrix(percent_df[3,c(1:3)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Wagle, MEK genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[3,c(1:3)])))
  abline(v=as.numeric(percent_df[3,3]), col="red",lty=2)
}

p_wagle

# Plot of Wagle et al., Mek
p_creighton %<a-% {
  barplot(rev(as.matrix(percent_df[4,c(1:3)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Creighton et al, AKT genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[4,c(1:3)])))
  abline(v=as.numeric(percent_df[4,3]), col="red",lty=2)
}

p_creighton


p_loboda %<a-% {
  barplot(rev(as.matrix(percent_df[5,c(1:3)])), horiz = TRUE, las = 2,
          xlim = c(0,4),
          main = "Loboda et al, RAS genes",
          xlab = "percentage",
          names.arg = rev(colnames(percent_df[5,c(1:3)])))
  abline(v=as.numeric(percent_df[5,3]), col="red",lty=2)
}

p_loboda


```


## Step 11: Create dotplots for enrichment - fold change and p-values and create dotplots for depletion - fold chang and p-values


```{r, echo = FALSE}

# Subset the enrichment p-values to those I will use in a figure:
matrix_of_Pvalues <- -log(Pvalue_df)[c(4,2,1,3,5),1]
ylabelnames <- rownames(Pvalue_df[c(4,2,1,3,5),])

# Gather the fold change data 
df_enrich_reorder <- df_enrich[c(4,2,1,3,5),1]
melted_foldEnrich <- melt(df_enrich_reorder, value.name = "fold_enrich")

# Reshape the enrichment p-values into a long format dataset:
long_Pvalues <- as.matrix(matrix_of_Pvalues) %>% 
  melt() %>%
  mutate(RNAseq_num = 1) %>%
  mutate(PubList = c(1:5)) %>%
  mutate(foldenrich = melted_foldEnrich$fold_enrich)

long_Pvalues$PubList <- as.factor(long_Pvalues$PubList)
colnames(long_Pvalues)[3] <- "neglogpvalue"


# Plot the data - enrichment
plot1 <- ggplot(long_Pvalues, aes(x=RNAseq_num, y=PubList, color=log(foldenrich))) + 
  geom_point(aes(size=neglogpvalue)) +
  scale_colour_gradient2(midpoint = 0, low = "navy", mid = "grey", high = "magenta", limits = c(-3, 3)) +
  scale_size(range = c(1, 20)) +
  scale_x_discrete(name = "Cohen-Sharir et al., gene list") +
  scale_y_discrete(name = "Published gene signature sets", labels=rev(ylabelnames), limits = c("5", "4", "3", "2", "1")) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), panel.grid.major = )

plot1

filename = paste("../03_output/", today, "_CohenSharir_dotplot.pdf", sep = "")
pdf(file = filename, width = 6, height = 4)
plot1
dev.off()


```






Session info:

```{r, echo = TRUE}
sessionInfo()
citation()
```

